skbeam/core/correlation.py

```diff
@@ -405,3 +409,356 @@ def auto_corr_scat_factor(lags, beta, relaxation_rate, baseline=1):
     return beta * np.exp(-2 * relaxation_rate * lags) + baseline
 
 
+def two_time_corr(labels, images, num_frames, num_bufs, num_levels=1):
+    """Wraps generator implementation of multi-tau two time correlation
+
+    This function computes two-time correlation
+    Original code : author: Yugang Zhang
+
+    See docstring for lazy_two_time
+    """
+    gen = lazy_two_time(labels, images, num_frames, num_bufs, num_levels)
+    for result in gen:
+        pass
+    return result.g2, result.lag_steps
+
+
+def lazy_two_time(labels, images, num_frames, num_bufs, num_levels=1,
+                  two_time_internal_state=None):
+    """ Generator implementation of two-time correlation
+
+    If you do not want multi-tau correlation, set num_levels to 1 and
+    num_bufs to the number of images you wish to correlate
+
+    Multi-tau correlation uses a scheme to achieve long-time correlations
+    inexpensively by downsampling the data, iteratively combining successive
+    frames.
+
+    The longest lag time computed is num_levels * num_bufs.
+    ** see comments on multi_tau_auto_corr
+
+    Parameters
+    ----------
+    labels : array
+        labeled array of the same shape as the image stack;
+        each ROI is represented by a distinct label (i.e., integer)
+    images : array
```

Comment:
Here you say that this parameter is an `array`, but in the explanation of the parameter you say it is an `iterable of 2D arrays`. If it can be an iterable, then this type should be `iterable`, not `array`